File: src/main/java-graphql/io/github/graphql/model/CreatedIssueContributionEdgeResponseProjection.java
package io.github.graphql.model;

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseField;
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseProjection;

/**
 * Response projection for CreatedIssueContributionEdge
 */
@javax.annotation.Generated(
    value = "com.kobylynskyi.graphql.codegen.GraphQLCodegen",
    date = "2021-03-06T19:47:48+0800"
)
public class CreatedIssueContributionEdgeResponseProjection extends GraphQLResponseProjection {

    public CreatedIssueContributionEdgeResponseProjection() {
    }

    @Override
    public CreatedIssueContributionEdgeResponseProjection all$() {
        return all$(3);
    }

    @Override
    public CreatedIssueContributionEdgeResponseProjection all$(int maxDepth) {
        this.cursor();
        if (projectionDepthOnFields.getOrDefault("CreatedIssueContributionEdgeResponseProjection.CreatedIssueContributionResponseProjection.node", 0) <= maxDepth) {
            projectionDepthOnFields.put("CreatedIssueContributionEdgeResponseProjection.CreatedIssueContributionResponseProjection.node", projectionDepthOnFields.getOrDefault("CreatedIssueContributionEdgeResponseProjection.CreatedIssueContributionResponseProjection.node", 0) + 1);
            this.node(new CreatedIssueContributionResponseProjection().all$(maxDepth - projectionDepthOnFields.getOrDefault("CreatedIssueContributionEdgeResponseProjection.CreatedIssueContributionResponseProjection.node", 0)));
        }
        this.typename();
        return this;
    }

    public CreatedIssueContributionEdgeResponseProjection cursor() {
        return cursor(null);
    }

    public CreatedIssueContributionEdgeResponseProjection cursor(String alias) {
        fields.add(new GraphQLResponseField("cursor").alias(alias));
        return this;
    }

    public CreatedIssueContributionEdgeResponseProjection node(CreatedIssueContributionResponseProjection subProjection) {
        return node(null, subProjection);
    }

    public CreatedIssueContributionEdgeResponseProjection node(String alias, CreatedIssueContributionResponseProjection subProjection) {
        fields.add(new GraphQLResponseField("node").alias(alias).projection(subProjection));
        return this;
    }

    public CreatedIssueContributionEdgeResponseProjection typename() {
        return typename(null);
    }

    public CreatedIssueContributionEdgeResponseProjection typename(String alias) {
        fields.add(new GraphQLResponseField("__typename").alias(alias));
        return this;
    }


}

File: src/main/kotlin-graphql/io/github/graphql/model/CreatedIssueContributionEdgeResponseProjection.kt
package io.github.graphql.model

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseField
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseProjection

/**
 * Response projection for CreatedIssueContributionEdge
 */
@javax.annotation.Generated(
    value = ["com.kobylynskyi.graphql.codegen.GraphQLCodegen"],
    date = "2021-03-06T19:48:04+0800"
)
open class CreatedIssueContributionEdgeResponseProjection : GraphQLResponseProjection() {

    override fun `all$`(): CreatedIssueContributionEdgeResponseProjection = `all$`(3)

    override fun `all$`(maxDepth: Int): CreatedIssueContributionEdgeResponseProjection {
        this.cursor()
        if (projectionDepthOnFields.getOrDefault("CreatedIssueContributionEdgeResponseProjection.CreatedIssueContributionResponseProjection.node", 0) <= maxDepth) {
            projectionDepthOnFields["CreatedIssueContributionEdgeResponseProjection.CreatedIssueContributionResponseProjection.node"] = projectionDepthOnFields.getOrDefault("CreatedIssueContributionEdgeResponseProjection.CreatedIssueContributionResponseProjection.node", 0) + 1
            this.node(CreatedIssueContributionResponseProjection().`all$`(maxDepth - projectionDepthOnFields.getOrDefault("CreatedIssueContributionEdgeResponseProjection.CreatedIssueContributionResponseProjection.node", 0)))
        }
        this.typename()
        return this
    }

    fun cursor(): CreatedIssueContributionEdgeResponseProjection = cursor(null)

    fun cursor(alias: String?): CreatedIssueContributionEdgeResponseProjection {
        fields.add(GraphQLResponseField("cursor").alias(alias))
        return this
    }

    fun node(subProjection: CreatedIssueContributionResponseProjection): CreatedIssueContributionEdgeResponseProjection = node(null, subProjection)

    fun node(alias: String?, subProjection: CreatedIssueContributionResponseProjection): CreatedIssueContributionEdgeResponseProjection {
        fields.add(GraphQLResponseField("node").alias(alias).projection(subProjection))
        return this
    }

    fun typename(): CreatedIssueContributionEdgeResponseProjection = typename(null)

    fun typename(alias: String?): CreatedIssueContributionEdgeResponseProjection {
        fields.add(GraphQLResponseField("__typename").alias(alias))
        return this
    }


}
